src/Common/src/System/Net/NetworkInformation/NetworkInformationException.cs

```diff
@@ -37,6 +36,10 @@ internal NetworkInformationException(SocketError socketError) : base((int)socket
         {
         }
 
+        internal NetworkInformationException(string message) : base(message)
```

Comment:
Design: In most cases for networking code, where it's very common to get exceptions we like to have the ability to differentiate the various errors via strong types (enum) or something easy to compare (an int). Not sure if this applies for the newly introduced exceptions. 

Either way, please ensure that an appropriate error code is set (maybe `E_FAIL` and definitely not `S_OK`).